spec/rex/parser/arguments_spec.rb

```diff
@@ -41,23 +43,96 @@
 
       it 'ignores unknown flags' do
         input = ['-a', 'action_name']
-        # Not sure if this flag dropping is intentional behavior
         expected_yields = [
+          # '-a' is dropped, 'action_name' is used as an argument
           [nil, 1, 'action_name'],
         ]
         expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
       end
 
       it 'ignores combined flags that do not exist' do
         input = ['-unknown-flags']
-        # Not sure if this flag dropping is intentional behavior
         expected_yields = [
-          ['-o', 0, nil],
+          [nil, 0, '-unknown-flags']
         ]
         expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
       end
     end
 
+    it 'parses a single long flag correctly' do
+      input = ['--help']
+      expected_yields = [
+        ['--help', 0, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses multiple long flags correctly' do
+      input = ['--help', '--jobs']
+      expected_yields = [
+        ['--help', 0, nil],
+        ['--jobs', 1, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses a long flag and short flag correctly' do
+      input = ['--help', '-h']
+      expected_yields = [
+        ['--help', 0, nil],
+        ['-h', 1, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses a short flag when Rex Arguments are in an array correctly' do
+      input = ['-o']
+      expected_yields = [
+        ['-o', 0, nil]
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses a long flag when in arguments array correctly' do
+      input = ['--options', 'option-arg']
+      expected_yields = [
+        ['--options', 0, 'option-arg']
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses multiple long flags when in an arguments array correctly' do
+      input = ['--quiet', '--options', 'sample-option']
+      expected_yields = [
+        ['--quiet', 0, nil],
+        ['--options', 1, 'sample-option']
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'parses a non-existent long flag correctly' do
+      input = ['--ultra-quiet']
+      expected_yields = [
+        [nil, 0, '--ultra-quiet']
+      ]
+      expect { |b| subject.parse(input, &b) }.to yield_successive_args(*expected_yields)
+    end
+
+    it 'prints usage in a sorted order correctly' do
```

Comment:
https://www.betterspecs.org/#describe 👀 

This needs to be in its own `#inspect` block, rather than being inside the `#parse` tests